scripts/lemonbar/header_bar.sh

```diff
@@ -67,6 +67,19 @@ while :; do
     buf="${buf} |%{A:gsimplecal&:} ⮖ $(clock)%{A}"
 
     echo $buf
-    # use `nowplaying scroll` to get a scrolling output!
-    sleep 0.5 # The HUD will be updated every second
+}
+
+while true; do
+
+  read -t 0.25 new_wm_content
```

Comment:
This polls from sysin, which in our case is xmonad piping workspace data and current application data. It integrates the latest data into the previous status data. To clarify

xmonad starts
  -> it launches this script and begins sysIn
  -> it formats the workspaces and current window automatically, including special formatting for `lemonbar`
  -> it passes that string to the script
  -> the script augments that data with other system information
  -> the script echos it out
  -> presumably, this script has been piped to lemonbar
  -> lemonbar eats that data, and renders it 
  -> (additionally, lemonbar should be piped to bash so commands can be executed from the lemonbar string)